### PR TITLE
Picard/collecthsmetrics

### LINKED ISSUE #793: new module: picard/CollectHsMetrics
I think it would be good to have a module for [picard/collecthsmetrics](https://broadinstitute.github.io/picard/picard-metric-definitions.html#HsMetrics)

- [x] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [x] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [x] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [x] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by projectoriented**
## PR checklist

Closes #793 
~~This PR needs the test-data from this PR (awaiting for review): https://github.com/nf-core/test-datasets/pull/359~~
- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag picard/collecthsmetrics --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by projectoriented** [comment]
Apologies for all the commit messages. I had to revert the version from `2.26.4` -> `2.26.2` due to conda issues below:

```
N E X T F L O W  ~  version 21.04.1
Launching `tests/modules/picard/collecthsmetrics/main.nf` [fervent_sanger] - revision: 7ececf13ed
[-        ] process > test_picard_collecthsmetrics:PICARD_COLLECTHSMETRICS -

[-        ] process > test_picard_collecthsmetrics:PICARD_COLLECTHSMETRICS -
Creating Conda env: bioconda::picard=2.26.4 [cache /home/mei.wu/tmp/pytest_workflow_dyq_gnsd/picard_collecthsmetrics_test_picard_collecthsmetrics/work/conda/env-2bd1fde7df90cefe945ae19f31c1dc35]

[-        ] process > test_picard_collecthsmetrics:PICARD_COLLECTHSMETRICS -

Error executing process > 'test_picard_collecthsmetrics:PICARD_COLLECTHSMETRICS (test)'

Caused by:
  Failed to create Conda environment
  command: conda create --mkdir --yes --quiet --prefix /home/mei.wu/tmp/pytest_workflow_dyq_gnsd/picard_collecthsmetrics_test_picard_collecthsmetrics/work/conda/env-2bd1fde7df90cefe945ae19f31c1dc35 bioconda::picard=2.26.4
  status : 1
  message:
    PaddingError: Placeholder of length '80' too short in package /home/mei.wu/tmp/pytest_workflow_dyq_gnsd/picard_collecthsmetrics_test_picard_collecthsmetrics/work/conda/env-2bd1fde7df90cefe945ae19f31c1dc35/bin/Rscript.
    The package must be rebuilt with conda-build > 2.0.
```

File: modules/picard/collecthsmetrics/main.nf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    input:
+    tuple val(meta), path(bam)
+    path fasta
+    path fai
+    path bait_intervals
+    path target_intervals
+
+    output:
+    tuple val(meta), path("*collecthsmetrics.txt"), emit: hs_metrics
+    path "versions.yml"                     , emit: versions
```
Comment (sguizard) [inline modules/picard/collecthsmetrics/main.nf:30]:
```suggestion
    tuple val(meta), path("*collecthsmetrics.txt"), emit: hs_metrics
    path "versions.yml"                           , emit: versions
```

File: modules/picard/collecthsmetrics/meta.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - bam:
```
Comment (sguizard) [inline modules/picard/collecthsmetrics/meta.yml:60]:
```suggestion
  - hs_metrics:
```

File: tests/modules/picard/collecthsmetrics/test.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+- name: picard collecthsmetrics test_picard_collecthsmetrics
+  command: nextflow run tests/modules/picard/collecthsmetrics -entry test_picard_collecthsmetrics -c tests/config/nextflow.config
+  tags:
+    - picard
+    - picard/collecthsmetrics
+  files:
+    - path: output/picard/test_collecthsmetrics.txt
```
Comment (sguizard) [inline tests/modules/picard/collecthsmetrics/test.yml:7]: The md5sum could'nt be computed?

**Review by sguizard — CHANGES_REQUESTED** [review]
Looks good to me overall. Just small modifications and one comment about the meta.yml

File: tests/modules/picard/collecthsmetrics/test.yml (hunk 3 above)
Comment (projectoriented) [reply tests/modules/picard/collecthsmetrics/test.yml:7]: Unfortunately not, it was too variable.

**Review by sguizard — APPROVED** [review]

**Comment by sguizard** [comment]
@projectoriented Thanks for the contribution!

**Pull request merged**